src/cascadia/WindowsTerminal/AppHost.cpp

```diff
@@ -1370,9 +1295,17 @@ void AppHost::_WindowMoved()
 void AppHost::_CloseRequested(const winrt::Windows::Foundation::IInspectable& /*sender*/,
                               const winrt::Windows::Foundation::IInspectable& /*args*/)
 {
-    const auto pos = _GetWindowLaunchPosition();
-    const bool isLastWindow = _windowManager.GetNumberOfPeasants() == 1;
-    _windowLogic.CloseWindow(pos, isLastWindow);
+    if (_windowManager.GetNumberOfPeasants() <= 1)
+    {
+        _quit();
+        return;
+    }
+
+    // Remove ourself from the list of peasants so that we aren't included in
+    // any future requests. This will also mean we block until any existing
+    // event handler finishes.
+    _windowManager.SignalClose(_peasant);
+    PostQuitMessage(0);

```

Comment:
Thanks! Bad rebase. 😕
It has to go to `AppHost::_CloseRequested`. `AppHost::_QuitRequested` doesn't need it, because the former is for window closing and the latter for when the application as a whole exits.